server/src/main/java/org/apache/iotdb/db/engine/compaction/TsFileManagement.java

```diff
@@ -214,6 +216,16 @@ public boolean merge(
       return false;
     }
 
+    if (unSeqMergeList.size() > maxOpenFileNumInEachUnseqCompaction) {
+      logger.info(
+          "{} too much unseq files to be merged, reduce it to {}",
+          storageGroupName,
+          maxOpenFileNumInEachUnseqCompaction);
+      unSeqMergeList =
+          unSeqMergeList.subList(
+              unSeqMergeList.size() - maxOpenFileNumInEachUnseqCompaction, unSeqMergeList.size());
+    }
```

Comment:
Older files should be merged first, otherwise, the overwritten data may re-emerge.

Reply:
The new unseq file may be more important?